starlette/datastructures.py

```diff
@@ -584,10 +576,11 @@ def __setitem__(self, key: str, value: str) -> None:
         set_key = key.lower().encode("latin-1")
         set_value = value.encode("latin-1")
 
-        found_indexes = []
-        for idx, (item_key, item_value) in enumerate(self._list):
-            if item_key == set_key:
-                found_indexes.append(idx)
+        found_indexes = [
```

Comment:
This is controversial. Not everybody believes it's an improvement. I'd prefer to not change it.